src/main/java/org/opensearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -309,11 +309,8 @@ public final void onFailure(Exception e) {
 
 	}
 
-	protected void saveAnUpdateConfigs(final Client client, final RestRequest request, final CType cType,
-									   final SecurityDynamicConfiguration<?> configuration, OnSucessActionListener<IndexResponse> actionListener) {
-		final IndexRequest ir = new IndexRequest(this.opendistroIndex);
-
-		//final String type = "_doc";
+	public static void saveAnUpdateConfigs(final String indexName, final Client client, final CType cType, final SecurityDynamicConfiguration<?> configuration, final ActionListener<IndexResponse> actionListener) {
```

Comment:
Did a small refactor on this method to make it reusable and cleaned up the instead name that was propagated throughout